### PR TITLE
ceph: add check for nil object status

### PR DESCRIPTION
**Pull request by Madhu-1**
in a couple of unit testing, the tests are panicking with nil pointer exception,This commit adds a check for status pointer before
operating on it.

the test was failing in https://jenkins.rook.io/blue/rest/organizations/jenkins/pipelines/rook/pipelines/rook/branches/PR-5982/runs/1/nodes/37/steps/38/log/?start=0
```
2020-08-05 15:24:58.226444 D | ceph-object-controller: creating s3 user "rook-ceph-internal-s3-user-checker-"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x158a1a1]

goroutine 52 [running]:
github.com/rook/rook/pkg/operator/ceph/object.updateStatusBucket(0x1c234e0, 0xc0005fdc20, 0x192c776, 0x9, 0x192b66d, 0x8, 0x192a378, 0x7, 0xc0006b8000, 0xa6)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/status.go:73 +0x361
github.com/rook/rook/pkg/operator/ceph/object.(*bucketChecker).checkObjectStore(0xc000619700, 0xc00060e180)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/health.go:85 +0x4b3
created by github.com/rook/rook/pkg/operator/ceph/object.(*ReconcileCephObjectStore).startMonitoring
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/controller.go:485 +0x32f
FAIL	github.com/rook/rook/pkg/operator/ceph/object	0.079s
=== RUN   TestPopulateDomainAndPort
```
Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
[test ceph]

[skip ci]

### CODE REVIEW

**Comment by Madhu-1** [comment]
just now I saw the actual fix is done at https://github.com/rook/rook/issues/5976#event-3625192359. how about a check for nil pointer?

**Comment by leseb** [comment]
> just now I saw the actual fix is done at [#5976 (comment)](https://github.com/rook/rook/issues/5976#event-3625192359). how about a check for nil pointer?

I think you fix is fine too.

File: pkg/operator/ceph/object/status.go [hunk 1]
```diff
@@ -69,7 +69,10 @@ func updateStatusBucket(client client.Client, name types.NamespacedName, phase c
 		logger.Warningf("failed to retrieve object store %q to update status to %v. %v", name, phase, err)
 		return
 	}
-
+	if objectStore.Status == nil {
```
Comment (travisn) [inline pkg/operator/ceph/object/status.go:72]: If the status isn't set yet, let's initialize it so it can be set for the first time.

Comment (Madhu-1) [reply pkg/operator/ceph/object/status.go:72]: @travisn initialize and continue or initialize and return?

Comment (travisn) [reply pkg/operator/ceph/object/status.go:72]: Initialize and continue. The struct just needs to be created the first time so the fields can be set.

Comment (Madhu-1) [reply pkg/operator/ceph/object/status.go:72]: Thanks for the suggestion.updated now PTAL

File: pkg/operator/ceph/object/status.go [hunk 2]
```diff
@@ -69,7 +69,10 @@ func updateStatusBucket(client client.Client, name types.NamespacedName, phase c
...
+		logger.Warningf("object store %q status is nil", name)
```
Comment (leseb) [inline pkg/operator/ceph/object/status.go:73]: I don't think we need to log here, let's just initialize.

Comment (Madhu-1) [reply pkg/operator/ceph/object/status.go]: done

**Comment by leseb** [comment]
CI hits https://github.com/rook/rook/issues/5996, known issue. Restarting.

**Pull request merged**